actions/setup-conftest/action.yaml

```
@@ -0,0 +1,51 @@
+name: Install conftest
+description: Install conftest
+
+inputs:
+  version:
+    description: |
+      Version of conftest to install.
+    default: 0.55.0
+
+runs:
+  using: composite
+
+  steps:
+    - name: Restore cache
+      id: restore
+      uses: actions/cache/restore@0c45773b623bea8c8e75f6c82b208c3cf94ea4f9 # v4.0.2
+      with:
+        path: ${{ github.workspace }}/bin
+        key: conftest-linux-amd64-${{ inputs.version }}
+
+    - name: Fetch Github Release Asset
+      id: fetch_asset
+      if: steps.restore.outputs.cache-hit != 'true'
+      uses: dsaltares/fetch-gh-release-asset@a40c8b4a0471f9ab81bdf73a010f74cc51476ad4 # 1.1.1
+      with:
+        repo: "open-policy-agent/conftest"
+        version: "tags/v${{ inputs.version }}"
+        file: "conftest_${{ inputs.version }}_Linux_x86_64.tar.gz"
```

can we make this work for other arches and potentially OSes? For ourselves, I would love the option to run on arm64 runners.

```suggestion
        file: "conftest_${{ inputs.version }}_${{ runner.os }}_${{ runner.arch }}.tar.gz"
```

---

Ah ok, I checked and it [looks like](https://github.com/open-policy-agent/conftest/blob/56d742d241174bbc790d7b6e6cb3a8632ffff576/.goreleaser.yml#L34) they're using something like `$(dpkg --print-architecture)` but with a special case for amd64.

I forgot but I had done this before. Code from an internal action (which could switch to `setup-conftest` 😁):

```bash
# Following upstream's behaviour:
# https://github.com/open-policy-agent/conftest/blob/56d742d241174bbc790d7b6e6cb3a8632ffff576/.goreleaser.yml#L34C36-L34C42
# So this workflow can run on arm64 or x86_64.
ARCH="$(dpkg --print-architecture)"
if [ "${ARCH}" = "amd64" ]; then
    ARCH="x86_64"
fi
```

---

That'll depend on a debian host... I'm having a look at a quick switch case in the steps.

---

As it's `goreleaser` it's probably `go env GOARCH` rather than `dpkg`, thinking about it

---

Tested on the internal branch, worked. :muscle: